torch/tensor.py

```diff
@@ -376,6 +376,12 @@ def split(self, split_size, dim=0):
         """
         if isinstance(split_size, int):
             return super(Tensor, self).split(split_size, dim)
+        elif isinstance(split_size, Tensor):
+            try:
```

Comment:
in this case it's a tensor of size 1?
Should we verify the size/dim instead of a try/except?

Reply:
The problem is that size is returned as a torch.Size object which is a tensor in itself, so we cannot compare that to an int. Dim on the other hand does return an int, but only how many dims it has so it could be a tensor of dim 1, but that does not mean it is a scalar. In the end this was the only way I have found that can reliable work in all cases.